src/quicer_stream.erl

```diff
@@ -120,15 +141,34 @@ start_link(Callback, Conn, StreamOpts) when is_atom(Callback) ->
           {error, Error :: {already_started, pid()}} |
           {error, Error :: term()} |
           ignore.
-start_link(Callback, Stream, Conn, StreamOpts, Props) when is_atom(Callback) ->
-    gen_server:start_link(?MODULE, [Callback, Stream, Conn, StreamOpts, Props], []).
+start_link(Callback, Stream, Conn, StreamOpts, Props)
+  when Callback =/= undefined
+       andalso is_atom(Callback)
+       andalso is_map(Props) ->
+    gen_server:start_link(?MODULE, [Callback, Stream, Conn, StreamOpts, Props, self()], []).
```

Comment:
Nit: since the number of arguments is large, maybe it makes sense to introduce a new record (or a map) type to pass the arguments to the init function?

This approach will also simplify init callback itself, since the record may contain default fields.

Reply:
yes. Agree.
 I will leave it to refactoring after functional work.